test/skate/settings/route_tab_test.exs

```diff
@@ -1,53 +1,66 @@
 defmodule Skate.Settings.RouteTabTest do
   use Skate.DataCase
+  import Skate.Factory
   alias Skate.Settings.RouteTab
-  alias Skate.Settings.Db.RouteTab, as: DbRouteTab
-  alias Skate.Settings.Db.User, as: DbUser
-
-  describe "create/1" do
-    test "creates and returns new record" do
-      route_tab = RouteTab.create("charlie")
-
-      assert %RouteTab{
-               preset_name: nil,
-               ladder_crowding_toggles: %{},
-               ladder_directions: %{},
-               selected_route_ids: []
-             } = route_tab
-
-      n_route_tabs = Skate.Repo.aggregate(DbRouteTab, :count)
-      assert n_route_tabs == 1
-      [sole_record] = Skate.Repo.all(DbRouteTab)
-
-      assert %{
-               preset_name: nil,
-               ladder_crowding_toggles: %{},
-               ladder_directions: %{},
-               selected_route_ids: []
-             } = sole_record
-
-      [sole_user] = Skate.Repo.all(DbUser)
-      assert %{username: "charlie"} = sole_user
-      assert sole_user.id == sole_record.user_id
-    end
-  end
 
   describe "get_all_for_user/1" do
     test "retrieves route tabs by username" do
-      RouteTab.create("user1")
-      RouteTab.create("user2")
+      route_tab =
+        build(:route_tab, %{preset_name: "some routes", selected_route_ids: ["1", "28"]})
+
+      RouteTab.update_all_for_user!("user1", [route_tab])
+      RouteTab.update_all_for_user!("user2", [route_tab])
 
       assert [%RouteTab{}] = RouteTab.get_all_for_user("user1")
     end
   end
 
-  describe "set/2" do
-    test "updates existing route_tab" do
-      route_tab = RouteTab.create("charlie")
+  describe "update_all_for_user!/2" do
+    test "adds a new tab entry" do
+      route_tab =
+        build(:route_tab, %{preset_name: "some routes", selected_route_ids: ["1", "28"]})
+
+      [_new_route_tab] = RouteTab.update_all_for_user!("charlie", [route_tab])
```

Comment:
Good idea. The one test where I still think it makes sense to omit such an assertion is for deleting, since the first half of that is just doing an insertion of a new record, a thing that's handled in the first test.